apps/system/lockscreen/js/lockscreen_inputpad.js

```diff
@@ -107,27 +98,14 @@
         // just need to reset Inputpad's internal state.
         this.states.passCodeEntered = '';
         this.states.passCodeErrorTimeoutPending = false;
-        this.updatePassCodeUI();
+        this.updatePassCodeUI(false);
         break;
       case 'lockscreen-inputappopened':
       case 'lockscreen-inputappclosed':
-        this.updatePassCodeUI();
+        this.updatePassCodeUI(false);
         break;
-      case 'click':
-        var key = evt.target.dataset.key;
-        if (!key &&
-            ('div' === evt.target.tagName.toLowerCase() &&
-             'a' === evt.target.parentNode.tagName.toLowerCase())
-           ) {
-          key = evt.target.parentNode.dataset.key;
-        }
-        if (!key) {
-          break;
-        }
-        // Cancel the default action of <a>
-        evt.preventDefault();
-        // handlePassCodeInput triggers updatePassCode
-        this.handlePassCodeInput(key);
+      case 'keyup':
+        this.handleKeyEvent(evt);
```

Comment:
Why not to register ``handleKeyEvent(evt)`` as an event listener directly on line 44 without putting it's invocation here?